src/MinresBlockSolver.cpp

```diff
@@ -20,311 +20,175 @@
 */
 
 #include "MinresBlockSolver.hpp"
-#include "utilities.hpp"
-#include <assert.h>
 
 namespace smoothg
 {
 
-/// implementation largely lifted from ex5p.cpp
-MinresBlockSolver::MinresBlockSolver(
-    MPI_Comm comm, mfem::HypreParMatrix* M, mfem::HypreParMatrix* D, mfem::HypreParMatrix* W,
-    const mfem::Array<int>& block_true_offsets,
-    bool use_W)
-    :
-    MixedLaplacianSolver(block_true_offsets),
-    minres_(comm),
-    comm_(comm),
-    use_W_(use_W),
-    operator_(block_true_offsets),
-    prec_(block_true_offsets)
+MinresBlockSolver::MinresBlockSolver(const MixedMatrix& mgl)
+    : MinresBlockSolver(mgl, {})
 {
-    Init(M, D, W);
 }
 
-void MinresBlockSolver::Init(mfem::HypreParMatrix* M, mfem::HypreParMatrix* D,
-                             mfem::HypreParMatrix* W)
+MinresBlockSolver::MinresBlockSolver(const MixedMatrix& mgl, const std::vector<int>& elim_dofs)
+    : MGLSolver(mgl), M_(mgl.GlobalM()), D_(mgl.GlobalD()), W_(mgl.GlobalW()),
```

Comment:
I think we need to do elimination on `M_` as well?

Reply:
For our examples, it doesn't seem to make a difference but yes it should be eliminated also.